python/ray/data/iterator.py

```diff
@@ -327,7 +339,7 @@ def iter_torch_batches(
                 if default_device.type != "cpu":
                     device = default_device
 
-            def collate_fn(batch: Union[np.ndarray, Dict[str, np.ndarray]]):
+            def finalize_fn(batch: Union[np.ndarray, Dict[str, np.ndarray]]):
                 return convert_ndarray_batch_to_torch_tensor_batch(
```

Comment:
The pytorch tensor creation and moving to the appropriate dtype should still be done as part of the collation since these are CPU operations. Only the host to device transfer should be done in finalize_fn with threadpool size of 1 since this uses GPU memory.

Reply:
Logic should be the following:

```
if collate_fn is not provided by user:
    collate_fn = torch tensor creation + moving to specified dtype
    finalize_fn = move tensors to gpu
if collate_fn is provided user:
    finalize_fn = None
```